src/app/components/harvest/pages/list/list.component.spec.ts

```diff
@@ -0,0 +1,145 @@
+    imports: [MockBawApiModule, SharedModule],
+    mocks: [ToastrService],
+  });
+
+  function setup(
+    project: Project,
+    mockHarvest: Harvest = new Harvest(generateHarvest({ status: "uploading" }))
+  ) {
+    spec = createComponent({
+      detectChanges: false,
+      data: {
+        project: { model: project },
+      },
+    });
+
+    const injector = spec.inject(Injector);
+    project["injector"] = injector;
+
+    const mockHarvestApi = spec.inject(HARVEST.token);
+    mockHarvest.addMetadata({
+      paging: { items: 1, page: 0, total: 1, maxPage: 5 },
+    });
+
+    // mock the harvest service filter API to populate the
+    // list component ngx-datatable
+    const mockResponse = of([mockHarvest]);
+    mockHarvestApi.filter.and.callFake(() => mockResponse);
+    spec.detectChanges();
+
+    // inject the NgbModal service so that we can
+    // dismiss all modals at the end of every test
+    modalService = spec.inject(NgbModal);
+    spec.detectChanges();
+
+    return mockHarvestApi;
+  }
+
+  function getAbortButton(): HTMLButtonElement {
+    return spec.debugElement.query(
+      (el) => el.nativeElement.innerText === "Abort"
+    ).nativeElement as HTMLButtonElement;
+  }
+
+  function getModalNextButton() {
+    return spec.query<HTMLButtonElement>(
+      "baw-harvest-confirmation-modal #next-btn",
+      { root: true }
+    );
+  }
+
+  function getModalCancelButton() {
+    return spec.query<HTMLButtonElement>(
+      "baw-harvest-confirmation-modal #cancel-btn",
+      { root: true }
+    );
+  }
+
+  beforeEach(() => {
+    defaultProject = new Project(generateProject());
+    defaultProject.addMetadata(generateProjectMeta({}));
+  });
+
+  afterEach(() => {
+    // dismiss all bootstrap modals, so if a test fails
+    // it doesn't impact future tests by using a stale modal
+    modalService.dismissAll();
+  });
+
+  it("should create", () => {
+    setup(defaultProject);
+    expect(spec.component).toBeInstanceOf(ListComponent);
+  });
+
+  it("should not change the status of a Harvest if the 'cancel' button is clicked in the abort modal", () => {
+    const harvestApi = setup(defaultProject);
+
+    // click the abort button
+    getAbortButton().click();
+    spec.detectChanges();
+
+    // click the cancel button
+    getModalCancelButton().click();
+    spec.detectChanges();
+
+    // assert that transitionHarvest was *not* called
+    expect(harvestApi.transitionStatus).not.toHaveBeenCalled();
+  });
+
+  it("should abort Harvest by changing status to 'complete' if the 'Abort Harvest' button is clicked in the abort modal", () => {
+    const harvestApi = setup(defaultProject);
+
+    // click the abort button
+    getAbortButton().click();
+    spec.detectChanges();
+
+    // click the cancel button
+    getModalNextButton().click();
+    spec.detectChanges();
+
+    // assert that transitionHarvest was *not* called
+    expect(harvestApi.transitionStatus).not.toHaveBeenCalledWith("complete");
```

Comment:
this is incorrect??

we want the harvest to transition to complete if the abort button was clicked. 
```suggestion
    // assert that transitionHarvest *was* called
    expect(harvestApi.transitionStatus).toHaveBeenCalledWith("complete");
```

Reply:
I have fixed this